### PR TITLE
Feature: Resumable training

### PR DESCRIPTION
**Pull request by hlfshell**
This PR introduces two methods of resumable training. The first is specifying a checkpoint load point that sets the model's weights to the checkpoints. This is good for resuming training from a pretrained model to add epochs. The second is the newer `BackupAndRestore` callback functionality. If the checkpoints folder exists w/ a per epoch checkpoint, it will resume training from the left-off epoch. This also remembers the prior loss and other training specific variables.

### CODE REVIEW

File: unet/train.py [hunk 1]
```diff
@@ -46,7 +52,8 @@ def train_unet(
     model.compile(optimizer="rmsprop", loss="sparse_categorical_crossentropy")
 
     callbacks = [
-        keras.callbacks.ModelCheckpoint("carla_segmentation.h5", save_best_only=True)
+        keras.callbacks.ModelCheckpoint(f"{checkpoint_directory}/unet.h5", save_best_only=True)
+        keras.callbacks.BackupAndRestore(backup_dir="./checkpoints/")
```
Comment (zekeflaton) [inline unet/train.py]: you can use the `checkpoint_directory` variable that gets passed in for this bit, right?

**Review by zekeflaton — DISMISSED** [review]
looks good.

File: unet/train.py (hunk 1 above)
Comment (hlfshell) [reply unet/train.py]: Yeah i probably should. Corrected.

**Review by Rhodwarrior — APPROVED** [review]

**Review by zekeflaton — APPROVED** [review]
